File: app/src/main/java/com/tt/handsomeman/DataSource.java
package com.tt.handsomeman;

import androidx.annotation.NonNull;
import androidx.paging.PageKeyedDataSource;

import com.tt.handsomeman.model.Job;

public class DataSource extends PageKeyedDataSource<Integer, Job> {
    @Override
    public void loadInitial(@NonNull LoadInitialParams<Integer> params, @NonNull LoadInitialCallback<Integer, Job> callback) {

    }

    @Override
    public void loadBefore(@NonNull LoadParams<Integer> params, @NonNull LoadCallback<Integer, Job> callback) {

    }

    @Override
    public void loadAfter(@NonNull LoadParams<Integer> params, @NonNull LoadCallback<Integer, Job> callback) {

    }
}

File: app/src/main/java/com/tt/handsomeman/DataSource.kt
package com.tt.handsomeman

import androidx.paging.PageKeyedDataSource

import com.tt.handsomeman.model.Job

class DataSource : PageKeyedDataSource<Int, Job>() {
    override fun loadInitial(params: PageKeyedDataSource.LoadInitialParams<Int>, callback: PageKeyedDataSource.LoadInitialCallback<Int, Job>) {

    }

    override fun loadBefore(params: PageKeyedDataSource.LoadParams<Int>, callback: PageKeyedDataSource.LoadCallback<Int, Job>) {

    }

    override fun loadAfter(params: PageKeyedDataSource.LoadParams<Int>, callback: PageKeyedDataSource.LoadCallback<Int, Job>) {

    }
}
